File: react-native-app/modules/face-processor/android/src/main/java/com/edgefacesentinel/faceprocessor/FaceProcessorModule.kt
/*
 * FaceProcessorModule.kt — Expo module whose only job is to register the
 * `faceProcessor` VisionCamera frame-processor plugin at app start. Expo runs
 * OnCreate once when the native module graph initialises, which is before the
 * camera mounts, so the plugin is ready when FaceCamera looks it up.
 */
package com.edgefacesentinel.faceprocessor

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import com.mrousavy.camera.frameprocessors.FrameProcessorPluginRegistry

class FaceProcessorModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("FaceProcessor")

    OnCreate {
      FrameProcessorPluginRegistry.addFrameProcessorPlugin("faceProcessor") { proxy, options ->
        FaceProcessorPlugin(proxy, options)
      }
    }
  }
}

File: react-native-app/modules/face-processor/ios/FaceProcessorModule.swift
/*
 * FaceProcessorModule.swift — Expo module that registers the `faceProcessor`
 * VisionCamera frame-processor plugin at startup (OnCreate). Registering from
 * the module avoids needing an ObjC `+load` category and an app bridging
 * header — the Swift plugin compiles inside this module's pod.
 */
import ExpoModulesCore
import VisionCamera

public class FaceProcessorModule: Module {
  public func definition() -> ModuleDefinition {
    Name("FaceProcessor")

    OnCreate {
      FrameProcessorPluginRegistry.addFrameProcessorPlugin("faceProcessor") { proxy, options in
        FaceProcessorPlugin(proxy: proxy, options: options)
      }
    }
  }
}
